pkg/backup/backup.go

```diff
@@ -174,94 +175,186 @@ func (b *Backup) Run() {
 	}
 }
 
+// saveSnap saves the latest snapshot if latest revision is greater than the given lastSnapRev.
 func (b *Backup) saveSnap(lastSnapRev int64) (int64, error) {
```

Comment:
probably should rename Backup to BackupManager to be more descriptive during this refactor.

Reply:
kk.